File: flutter/lib/api/wildbird_deaths_api.dart
import 'package:intl/intl.dart';
import 'package:livetec_flutter_app/api/fetch_api.dart';
import 'package:livetec_flutter_app/types/death.dart';

Future<List<Death>> getWildbirdDeaths(DateTime from, DateTime to) async {
  final data = await fetchApi(
    FetchMethod.get,
    '/api/wildbird-deaths',
    query: {
      'from': DateFormat('yyyy-MM-dd').format(from),
      'to': DateFormat('yyyy-MM-dd').format(to),
    },
  );

  if (data == null) return [];
  return (data as List).map((element) => Death.fromJson(element)).toList();
}

File: react/src/api/wildbird-deaths-api.ts
import type { Death } from '../types/death';
import { fetchApi } from './fetch-api';
import type { Dayjs } from 'dayjs';

export const getWildbirdDeaths = (from: Dayjs, to: Dayjs, species?: string) =>
  fetchApi<Death[]>('GET', '/api/wildbird-deaths', {
    from: from.format('YYYY-MM-DD'),
    to: to.format('YYYY-MM-DD'),
    species,
  });
